components/apimgt/org.wso2.carbon.apimgt.core/src/main/java/org/wso2/carbon/apimgt/core/dao/impl/ApiFileDAOImpl.java

```diff
@@ -0,0 +1,486 @@
+        APIFileUtils.writeObjectAsJsonToFile(endpoint, endpointFileLocation);
+
+        if (log.isDebugEnabled()) {
+            log.debug("Successfully saved endpoint  definition for endpoint: " + endpoint.getName());
+        }
+    }
+
+    /**
+     * write the given API gateway config to file system
+     *
+     * @param config gateway config of the api
+     * @param api {@link API} instance
+     * @param exportLocation file system location to write the API gateway config.
+     * @throws APIMgtDAOException if an error occurs while writing the API definition
+     */
+    private void exportGatewayConfigToFileSystem(String config, API api, String exportLocation)
+            throws APIMgtDAOException {
+
+        if (config == null) {
+            // not gateway config found, return
+            log.info("No gateway configuration found for API with api: " + api.getName() + ", version: " + api
+                    .getVersion());
+            return;
+        }
+
+        String gatewayConfigLocation = exportLocation + File.separator + GATEWAY_CONFIGURATION_DEFINITION_FILE;
+        APIFileUtils.createFile(gatewayConfigLocation);
+        APIFileUtils.writeToFile(gatewayConfigLocation, config);
+        if (log.isDebugEnabled()) {
+            log.debug("Successfully exported gateway configuration for api: " + api.getName() + ", version: " + api
+                    .getVersion());
+        }
+    }
+
+    /**
+     * write the given Endpoint definition to file system
+     *
+     * @param swaggerDefinition swagger definition
+     * @param api {@link API} instance relevant to the swagger definition
+     * @param exportLocation file system location to which the swagger definition will be written
+     * @throws APIMgtDAOException if an error occurs while writing the Endpoint
+     */
+    private void exportSwaggerDefinitionToFileSystem(String swaggerDefinition, API api, String exportLocation)
+            throws APIMgtDAOException {
+        String swaggerDefinitionLocation =
+                exportLocation + File.separator + SWAGGER_DEFINITION_FILE_NAME + api.getId() + JSON_EXTENSION;
+        APIFileUtils.createFile(swaggerDefinitionLocation);
+        APIFileUtils.writeStringAsJsonToFile(swaggerDefinition, swaggerDefinitionLocation);
+
+        if (log.isDebugEnabled()) {
+            log.debug("Successfully exported Swagger definition for api: " + api.getName() + ", version: " + api
+                    .getVersion());
+        }
+    }
+
+    private Object fetchObject(File file, Class c, FilenameFilter filenameFilter) {
+        File[] files = (filenameFilter != null) ? file.listFiles(filenameFilter) : file.listFiles();
+        if (files != null && files.length > 0) {
+            return constructObjectSummaryFromFile(files[0].getAbsolutePath(), c);
+        } else if (!file.isDirectory()) {
+            constructObjectSummaryFromFile(file.getAbsolutePath(), c);
```

Comment:
If we skip return here, it will always return null when comes to the else if block. Shall we correct that?